### PR TITLE
Bugfix for minimum needs:

### PR DESCRIPTION
**Pull request by lucernae**
- Hide the close button when adding resource
- Setup the parameter container UI so it loads parameter widgets
- Make the profile combo box and minimum needs list responds when it first loads

### CODE REVIEW

File: safe/gui/tools/minimum_needs/needs_manager_dialog.py [hunk 1]
```diff
@@ -197,9 +198,7 @@ def __init__(self, parent=None, dock=None):
         self.profile_combo.activated.connect(self.select_profile)
         # noinspection PyUnresolvedReferences
         self.stacked_widget.currentChanged.connect(self.page_changed)
-        # initial sync profile_combo and resource list
-        self.clear_resource_list()
-        self.populate_resource_list()
```
Comment (lucernae) [inline safe/gui/tools/minimum_needs/needs_manager_dialog.py]: This two line is redundant with line 192-193.
Is it okay to delete it?

File: safe/gui/tools/minimum_needs/needs_manager_dialog.py [hunk 2]
```diff
@@ -321,6 +320,8 @@ def add_new_resource(self):
             "and no more than {{ Maximum allowed }}. This should be provided "
             "{{ Frequency }}."))
         self.stacked_widget.setCurrentWidget(self.resource_edit_page)
+        # hide the close button
+        self.button_box.button(QDialogButtonBox.Close).setHidden(True)
```
Comment (lucernae) [inline safe/gui/tools/minimum_needs/needs_manager_dialog.py]: Hide the close button when we add resources

File: safe/gui/tools/minimum_needs/needs_manager_dialog.py [hunk 3]
```diff
@@ -511,7 +513,7 @@ def remove_resource(self):
         """Remove the currently selected resource.
         """
         self.mark_current_profile_as_pending()
-        for item in self.resgiources_list.selectedItems():
+        for item in self.resources_list.selectedItems():
```
Comment (lucernae) [inline safe/gui/tools/minimum_needs/needs_manager_dialog.py]: Stupid typo committed previously :p

**Comment by lucernae** [comment]
@akbargumbira or @ismailsunni  can you review it and merge it quickly. The bug is causing the minimum needs add/edit resources can't be loaded.

**Comment by ismailsunni** [comment]
Why do you need to change test data?

**Comment by lucernae** [comment]
I reverted the test file now

**Pull request merged**